pkg/operator/ceph/cluster/mon/spec.go

```diff
@@ -58,6 +58,13 @@ func (c *Cluster) getLabels(monConfig *monConfig, canary, includeNewLabels bool)
 		if monConfig.Zone != "" {
 			labels["zone"] = monConfig.Zone
 		}
+		// Populate provided labels if they do not exist already
+		for k, v := range monVolumeClaimTemplate.ObjectMeta.Labels {
```

Comment:
This loop needs to go inside the line 53 conditional that checks if the `monVolumeClaimTemplate != nil`. The CI is failing wherever the template is not set, due to this issue.